[Service Offloading] Key/Mouse Input Support

Chikdolman opened the pull request:
** Note that Android JAVA needs to be migrated **

Support Keyboard and mouse input.
Only support windows event.
For Android(Galaxy) Knox must internally be supported.
To enable Galaxy Knox key event, following must be set in args.gn
1. target_os = "android"
2. enable_service_offloading = true
3. enable_service_offloading_knox = true

Sunwoo-Nam commented on build/config/BUILD.gn:
```diff
@@ -141,6 +141,7 @@ config("feature_flags") {
   if (enable_service_offloading) {
     defines += [ "SERVICE_OFFLOADING" ]
   }
+
```
Please remove this line.

Chikdolman replied:
Done

Sunwoo-Nam commented on third_party/blink/renderer/modules/input_control/BUILD.gn:
```diff
@@ -0,0 +1,62 @@
+} else if(is_android && enable_service_offloading_knox){
+    blink_modules_sources("input_control") {
+      deps = [ 
+        "//base", 
```
Duplicate code should be placed outside the conditional statement.

Chikdolman replied:
Code is modified. Thanks

Sunwoo-Nam commented on third_party/blink/renderer/modules/input_control/BUILD.gn:
```diff
@@ -0,0 +1,62 @@
+      ]
+
+      sources = [
+        "input_control.h",
```
Duplicate code should be placed outside the conditional statement.

Chikdolman replied:
Code is modified. Thanks

Sunwoo-Nam commented on third_party/blink/renderer/modules/input_control/BUILD.gn:
```diff
@@ -0,0 +1,62 @@
+        "input_control_android.cc"
+      ]
+
+      if (is_android && enable_service_offloading_knox) {
```
The conditional statement is duplicated.

Chikdolman replied:
Code is modified. Thanks

Sunwoo-Nam commented on third_party/blink/renderer/modules/modules_idl_files.gni:
```diff
@@ -966,6 +970,7 @@ if (!is_android) {
       get_path_info([
                       "serial/navigator_serial.idl",
                       "serial/worker_navigator_serial.idl",
+                      "input_control/window_input_call.idl",
```
Doesn't this need |enable_service_offloading|?

Chikdolman replied:
Code is modified. Thanks

Sunwoo-Nam commented on third_party/blink/renderer/modules/BUILD.gn:
```diff
@@ -190,6 +190,11 @@ target("jumbo_" + modules_target_type, "modules") {
     cflags = [ "/wd4334" ]  # Result of 32-bit shift implicitly converted to 64 bits.
   }
 
+  if ((is_android && enable_service_offloading_knox) || is_win) {
```
Doesn't this need |enable_service_offloading|?

Chikdolman replied:
Done

Sunwoo-Nam reviewed (commented):
In the commit message,
"2. enable_service_offloading_knox = false"
I think 'false' should be changed to 'true'.

Sunwoo-Nam commented on third_party/blink/renderer/modules/BUILD.gn:
```diff
@@ -190,6 +190,11 @@ target("jumbo_" + modules_target_type, "modules") {
+  if (((is_android && enable_service_offloading_knox) || is_win) && enable_service_offloading) {
+    sub_modules += ["//third_party/blink/renderer/modules/input_control"]
+  }
+
```
Unnecessary line.

Chikdolman replied:
Done

Sunwoo-Nam reviewed (commented):
Please check white spaces.

Sunwoo-Nam reviewed (commented):
Please adjust the indentations.

Sunwoo-Nam reviewed (approved):
LGTM

Chikdolman committed:
[Service Offloading] Key/Mouse Input Support

Support Keyboard and mouse input for windows only.
To enable Galaxy Knox key event, Knox must internally be supported and following must be set in args.gn
1. target_os = "android"
2. enable_service_offloading = true
3. enable_service_offloading_knox = true

Chikdolman committed:
[Service Offloading] Adds variable to AndroidManifest
variables are added to the manifest for Java
1. enable_service_offloading
2. enable_service_offloading_knox
